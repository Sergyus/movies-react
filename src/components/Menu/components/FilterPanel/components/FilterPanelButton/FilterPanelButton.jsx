import React, { useCallback } from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import styles from '../../FilterPanel.scss';

const FilterPanelButton = ({
  name, setActiveItem, setActive, fetchMovies, activeItem,
}) => (
  <li className={styles.navItem}>
    <button
      disabled={activeItem === name}
      aria-label={name}
      type="button"
      className={cn(styles.btn, setActiveItem(name))}
      onClick={useCallback(() => {
        setActive(name);
        fetchMovies(1);
      }, [fetchMovies, name, setActive])}
    >
      {name}
    </button>
  </li>
);

FilterPanelButton.propTypes = {
  name: PropTypes.string.isRequired,
  activeItem: PropTypes.string,
  setActive: PropTypes.func.isRequired,
  setActiveItem: PropTypes.func.isRequired,
  fetchMovies: PropTypes.func,
};

FilterPanelButton.defaultProps = {
  fetchMovies: () => {},
  activeItem: 'Trending',
};

export default FilterPanelButton;
