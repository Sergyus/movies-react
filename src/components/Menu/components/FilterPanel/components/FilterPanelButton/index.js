import FilterPanelButton from './FilterPanelButton';

export default FilterPanelButton;
