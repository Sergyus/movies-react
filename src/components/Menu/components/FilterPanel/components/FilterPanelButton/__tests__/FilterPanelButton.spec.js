import React from 'react';
import { create, act } from 'react-test-renderer';
import FilterPanelButton from '..';

describe('FilterPanel component', () => {
  const activeItem = 'Trending';
  const setActiveItem = jest.fn();
  const setActive = jest.fn();
  it('FilterPanel component renders correctly', () => {
    const tree = create(<FilterPanelButton activeItem={activeItem} setActiveItem={setActiveItem} name="Trending" setActive={setActive} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('Set active called 1 time', () => {
    const tree = create(<FilterPanelButton activeItem={activeItem} setActiveItem={setActiveItem} name="Top Rated" setActive={setActive} />);
    const button = tree.root.findByProps({ 'aria-label': 'Top Rated' });
    act(() => {
      button.props.onClick();
    });
    expect(setActive.mock.calls.length).toEqual(1);
  });

  it('Set active item called 3 times', () => {
    const tree = create(<FilterPanelButton activeItem={activeItem} setActiveItem={setActiveItem} name="Coming Soon" setActive={setActive} />);
    const button = tree.root.findByProps({ 'aria-label': 'Coming Soon' });
    act(() => {
      button.props.onClick();
    });
    expect(setActiveItem.mock.calls.length).toEqual(3);
  });
});
