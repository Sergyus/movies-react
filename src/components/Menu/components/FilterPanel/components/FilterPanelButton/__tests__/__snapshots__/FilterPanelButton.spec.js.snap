// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterPanel component FilterPanel component renders correctly 1`] = `
<li
  className="navItem"
>
  <button
    aria-label="Trending"
    className="btn"
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    Trending
  </button>
</li>
`;
